augur/parse.py

```diff
@@ -143,6 +142,8 @@ def register_parser(parent_subparsers):
     parser.add_argument('--sequences', '-s', required=True, help="sequences in fasta or VCF format")
     parser.add_argument('--output-sequences', required=True, help="output sequences file")
     parser.add_argument('--output-metadata', required=True, help="output metadata file")
+    parser.add_argument('--output-id-field', required=False,
+                        help="The record field to use as the sequence identifier in the FASTA output.")
```

Comment:
Suggestion: Describe the default behavior, e.g.

> If not provided, this will use the first available of {DEFAULT_ID_COLUMNS}. If none of those are available, it will use the first field.

Also would be good to add tests for this default behavior (before the change, so it's apparent that the order of preference changed from `name, strain` to `strain, name`). Technically this is a breaking change.

Reply:
Good point, I've emphasized the breaking change of order preference `name,strain` to `strain,name` by:

1. Adding a test for `name,strain` order https://github.com/nextstrain/augur/pull/1403/commits/ba2f9883422cfcc5b9d8118c47eb13e20c3cf6d9
2. Documenting the switch to use `DEFAULT_ID_COLUMNS` which results in breaking change `strain,name` https://github.com/nextstrain/augur/pull/1403/commits/8dc06946e3fb045d84f03908c9941e965b22daaf
3. Then applying the rest of the `--output-id-field` commits

Further suggestions welcome, mostly trying to reorganize the edits for ease of code-reading

Reply:
And applied the suggested edit for the help message https://github.com/nextstrain/augur/pull/1403/commits/6ac73c0d6d6742a2c49fbe7a8e4378d9025a9821 , good catch! 
This PR is ready for review